Allow easy host change

cristian-aldea opened the pull request:
closes #217 

The host.env.ts file was added to allow the developer to easily change the hostname where their application is being served, to allow for use cases where they don't want to run on localhost. For example, if they want other devices on their local network to be able to access the application, using localhost won't work, and they would need to host in on their machine's IP address for example

The README was updated to include info on how this mechanism works

The reason I went with a .ts file instead of adding these variables to our .env file is that the frontend cannot read the environment variables of our web server, since it is served to the client's browser. My solution, therefore, involved creating a ts in our library, which would allow us to set these values for both apps without hardcoding them.

How to test:
1. Follow instructions in the readme
2. Set your hostname to localhost and make sure everything works as before
3. Set your hostname to something else (for example, your machine's IP address) and test if you can access the app from other devices). Don't forget the --host flag!

cristian-aldea committed:
fix(hostname): Allow easy hostname change through host.env.ts

The host.env.ts file was added to allow the developer to easily change the hostname where their application is being served, to allow for usecases where they don't want to run on localhost. For example, if they want other devices on their local network to be able to acce
ss the application, using localhost won't work, and they would need to host in on their machine's ip address for example

The README was updated to include info on how this mechanism works

The reason I went with a .ts file instead of adding these variables to our .env file is that the frontend cannot read the environment variables of our webserver, since it is served to the client's browser. My solution therefore involved gitignoring a ts with the values, which would allow us to set these values without hardcoding them.

Ref #217

cristian-aldea committed:
fix(test/lint): Fix test/lint errors

ref #217

osahtout commented on README.md:
````diff
@@ -79,9 +79,26 @@ Some secrets and sensitive configuration files are needed to properly operate ce
+   }
+   ```
+
+1. Add a `host.env.ts` under the `libs/types/src/` folder. This file specify the hostnames where the application are currently being hosted, and makes it easy to change them.
+
+   ```ts
+   export const UI_HOSTNAME = 'hostname';
+   export const API_HOSTNAME = 'hostname';
+   ```
````
is there a particular reason why these are not added to .env file? I'm not sure there's a need to add another file.

osahtout replied:
I also noticed that all but one uses http://. I think we can set it by default to be http://localhost: in the .env file and developers can change the localhost to what ever they want without the need to add another file.

osahtout replied:
Oh dang, I must have just skipped that part of the description sorry. 

can I ask why do we need to ignore the ts file if we just set the two string to 'http://localhost' ?

cristian-aldea replied:
As discussed on slack, I'll be going with your suggestion @osahtout. I'll be removed the file from gitignore so that other don't need to set it up. We'll just need to keep in kind to not commit changes to that file.

cristian-aldea replied:
However, I think its fine to keep the protocol out of the name (http://), since its not what I want to make configurable, just  the host name

osahtout commented on apps/mull-ui/src/app/components/event-card/event-card.spec.tsx:
```diff
@@ -27,12 +26,6 @@ describe('EventCard', () => {
     expect(baseElement).toBeTruthy();
   });
 
-  it('should match snapshot', () => {
```
wait why was this removed?

cristian-aldea replied:
Will be putting this back 🚀

r-saba commented on README.md:
```diff
@@ -79,9 +79,26 @@ Some secrets and sensitive configuration files are needed to properly operate ce
-   - `nx serve mull`
+   - `nx serve mull-ui`
+   - If you specified a host other than `localhost` in your `host.env.ts`, then make sure to add the `--host` flag so that the webserver can be configured properly to serve the app on the right host
+     - example: `nx sere mull-ui --host 123.123.123.123`
```
```suggestion
     - example: `nx serve mull-ui --host 123.123.123.123`
```

cristian-aldea replied:
thanks for looking out! Sorry for the sloppy readme updates on my end 😅

r-saba commented on README.md:
````diff
@@ -79,9 +79,26 @@ Some secrets and sensitive configuration files are needed to properly operate ce
+   }
+   ```
+
+1. Add a `host.env.ts` under the `libs/types/src/` folder. This file specify the hostnames where the application are currently being hosted, and makes it easy to change them.
````
```suggestion
1. Add a `host.env.ts` under the `libs/types/src/` folder. This file specifies the hostnames where the application are currently being hosted, and makes it easy to change them.
```

r-saba reviewed (approved):
LGTM! Works both with localhost and my computers ip address. Able to access the local website from other devices when using my ipv4 address. 

Left some documentation corrections to fix before merging.

cristian-aldea committed:
fix(host env): Implement omar suggestion

Removed the host.env.ts file from gitignore

Added back snapshot tests that were removed.

Reverted changes to e2e tests

cristian-aldea committed:
doc(nx): Remove unnecessary NX readme

ref #217

RGPosadas reviewed (approved):
Works perfectly. LGTM!
